rest-api/dao/participant_summary_dao.py

```diff
@@ -150,6 +150,48 @@ def _get_dna_isolates_sql_and_params():
       params
   )
 
+def _get_sample_status_time_sql_and_params():
+  """Gets SQL and params that to update enrollmentStatusCoreStoredSampleTime field
+  on the participant summary.
+  """
+
+  tests_sql, params = get_sql_and_params_for_array(
+    config.getSettingList(config.DNA_SAMPLE_TEST_CODES), 'dna')
+
+  confirm_value_sql = '%s' % ','.join(["""COALESCE((SELECT max(confirmed) FROM biobank_stored_sample
+          WHERE biobank_stored_sample.biobank_id = participant_summary.biobank_id
+          AND biobank_stored_sample.confirmed IS NOT NULL
+          AND biobank_stored_sample.test = '%s'),'3000-01-01')""" % params[key] for key in params])
+
+  confirm_time_sql = """
+  (
+    SELECT 
+      CASE
+        WHEN stored_time='3000-01-01' THEN NULL
+        ELSE stored_time
+      END
+    FROM    
+      (SELECT
+        MIN({confirm_value_sql}) as stored_time
+      FROM
+        biobank_stored_sample
+      WHERE
+        biobank_stored_sample.biobank_id = participant_summary.biobank_id
+        AND biobank_stored_sample.confirmed IS NOT NULL
+        AND biobank_stored_sample.test IN {tests_sql}
+      ) X  
+  )
+  """.format(confirm_value_sql=confirm_value_sql, tests_sql=tests_sql)
+
+  sql = """
+    UPDATE
+      participant_summary
+    SET
+      enrollment_status_core_stored_sample_time = {confirm_time_sql}
```

Comment:
We may want to have a WHERE clause so that this only gets set for participant summaries where it hasn't been set already. 

You may also be able to combine this with the enrollment status updating SQL... not sure if it matters.

Reply:
Yeah, that rename is fine by me. (A little less explicit, but the official definition of "core" is based on stored sample time, so this is fine.)